src/bundle/Resources/public/js/alloyeditor/src/buttons/ez-btn-anchoredit.js

```diff
@@ -0,0 +1,164 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import AlloyEditor from 'alloyeditor';
+
+const CLASS_HAS_ANCHOR = 'ez-has-anchor';
+
+export default class EzBtnAnchorEdit extends Component {
+    constructor(props) {
+        super(props);
+
+        this.updateValue = this.updateValue.bind(this);
+        this.saveAnchor = this.saveAnchor.bind(this);
+        this.removeAnchor = this.removeAnchor.bind(this);
+
+        this.getStateClasses = AlloyEditor.ButtonStateClasses.getStateClasses;
+
+        this.block = null;
+
+        this.state = {
+            value: '',
+        };
+    }
+
+    componentDidMount() {
+        const block = this.findBlock();
+        const value = block.getId();
+
+        if (block.hasClass('ez-icon--anchor')) {
+            this.renderIcon();
+        }
+
+        this.setState(() => ({ value }));
+    }
+
+    static get key() {
+        return 'ezanchoredit';
+    }
+
+    findBlock() {
+        const nativeEditor = this.props.editor.get('nativeEditor');
+        const focused = nativeEditor.widgets.focused;
+        const path = nativeEditor.elementPath();
+        let block = path.block;
+
+        if (this.block) {
+            return this.block;
+        }
+
+        if (!block && focused) {
+            block = focused.element;
+        }
+
+        if (block && block.is('li')) {
+            block = block.getParent();
+        }
+
+        if (!block && path.contains('table')) {
+            block = path.elements.find((element) => element.is('table'));
+        }
+
+        this.block = block;
+
+        return block;
+    }
+
+    findIcon() {
+        const block = this.findBlock();
+
+        return [...block.getChildren().$].find((child) => child.classList && child.classList.contains('ez-icon--anchor'));
+    }
+
+    updateValue({ nativeEvent }) {
+        const value = nativeEvent.target.value;
+
+        this.setState(() => ({ value }));
+    }
+
+    removeAnchor() {
+        const block = this.findBlock();
+        const icon = this.findIcon();
```

Comment:
Nitpick: we can pass `block` to  `findIcon` instead of finding it twice.

Reply:
But I wanted to have a more atomic method. And if you look at`findBlock()` you will find out that the method is calculating things only one time, the second one is returning block immediately.

Reply:
You're right, I didn't notice `this.block = block;`. I like your approach! :)